include/openssl/core_names.h

```diff
@@ -182,6 +183,29 @@ extern "C" {
 #define OSSL_PKEY_PARAM_DIGEST_SIZE         "digest-size"
 #define OSSL_PKEY_PARAM_MGF1_DIGEST         "mgf1-digest"
 #define OSSL_PKEY_PARAM_MGF1_PROPERTIES     "mgf1-properties"
+/*
+ * These are used to get diverse DER encoded AlgorithmIdentifiers, conforming
+ * to RFC 3279 with updates as well as other similar standards.  They may be
+ * used with a KEYMGMT to get the Subject Public Key Algorithm as well as
+ * with diverse other PKEY operations, such as getting a Signature Algorithm
+ * from a SIGNATURE operation.
+ *
+ * The value is expected to be an octet string with the DER encoded
+ * AlgorithmIdentifier for that operation or keypair.
+ *
+ * Supporting this parameter is not mandatory, but will determine whether
+ * it can be used for certain purposes.
```

Comment:
Nit: This phrasing seems a bit vague (and I get this is intentional). I think it might benefit both us in the future and authors of external providers if we at least add some kind of example to give an idea of what kind of purposes would benefit from enabling this.

My proposal would be for example to add, just before the end `(e.g., certain CMS operations might require support of this parameter or OSSL_CMS_PARAM_ALGORITHM_ID, as described further down).`

Reply:
Added